add additional details about partial sort (#17)
justinmeiners/efficient-programming-with-components: 04_instrumented.md

- [`std::partial_sort`](https://en.cppreference.com/w/cpp/algorithm/partial_sort)

    What's the interface?
    It takes three iterators `first`, `middle`, and `last`.
    What does it do?

>     It orders the elements so that `first` to `middle`
>     contains the smallest elements from the entire range.
>     Then it sorts those first elements.

Reviewer: This seems to imply that there are two distinct phases to the algorithm. Perhaps just shorten it to this?
```suggestion
    It orders the elements so that `first` to `middle`
    contains the smallest elements from the entire range, in sorted order.
```

    For example, suppose you give it 100 elements and you want to sort from 1st, to the 10th, to the last.
    The smallest 10 will be moved to the front in sorted order.

>     But the last 90 elements will be left in *some order*.

Reviewer: ```suggestion
    But the last 90 elements will be left in *some unspecified order*.
```
